reframe/core/pipeline.py

```diff
@@ -421,6 +421,14 @@ class RegressionTest(metaclass=RegressionTestMeta):
     use_multithreading = fields.TypedField('use_multithreading',
                                            bool, type(None))
 
+    #: Specify the maximum time the job can be pending before starting running
+    #:
+    #: The time is specified with the same format of `time_limit`.
+    #:
+    #: :type: :class:`tuple[int]`
+    #: :default: :class:`None
+    max_pending_time = fields.TimerField('max_pending_time', type(None))
```

Comment:
I don't like very much the tuple syntax of `TimerField` , so I suggest fixing #1106 first before proceeding with this PR. Because otherwise we introduce a new variable with the same bad syntax.